### PR TITLE
E2E: Add scaffolding for rollkit E2E

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #5992

This PR Sets up scaffolding for a test which 

- deploys a celestia dev net
- deploys a wasm simapp
- deploys a rollkit app

Note: the underlying containers are deployed successfuly, but the E2E test will still fail due to interchaintest expecting ports to remain default, this can be addressed in a follow up.

Due to the nature of the test, some internal functions from interchaintest were copied over and used in order to do some lower level docker operations such as copying over private keys and manually creating a container (for celestia).

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/tests/rollkit/rollkit_test_util.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
+package rollkit
+
+// NOTE: this file should be able to be removed when https://github.com/strangelove-ventures/interchaintest/issues/869 is resolved
```
Comment (chatton) [inline e2e/tests/rollkit/rollkit_test_util.go:3]: :point_up:

Comment (chatton) [reply e2e/tests/rollkit/rollkit_test_util.go:3]: this is all just copied from interchaintest

File: e2e/tests/rollkit/rollkit_test.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+		options.ChainASpec.ChainID = "rollkit-app"
+		options.ChainASpec.Bin = "gmd"
+		options.ChainASpec.Bech32Prefix = "gm"
+		options.ChainASpec.AdditionalStartArgs = []string{
+			"--rollkit.aggregator",
+			fmt.Sprintf("--rollkit.da_address=celestia:26650"),
+			fmt.Sprintf("--rollkit.da_start_height=%s", s.getCelestiaBlockHeight(celestiaHostPort)),
+			"--minimum-gas-prices=0.025atoma",
+
+			// NOTE: the following feels shouldn't be required, but the rollkit app fails to communicate with
```
Comment (Reecepbcups) [inline e2e/tests/rollkit/rollkit_test.go:67]: ref: https://github.com/strangelove-ventures/interchaintest/pull/952#issuecomment-2010473851 may help here using `HostPortOverride`

Comment (chatton) [reply e2e/tests/rollkit/rollkit_test.go:67]: ah looks perfect, thank you!

**Comment by chatton** [comment]
@crodriguezvega I don't think we need to get this E2E merged into main until we have a working contract, I'll remove the priority tag, if you disagree feel free to re-add!

File: e2e/tests/rollkit/rollkit_test_util.go (hunk 1 above)
Comment (Reecepbcups) [reply e2e/tests/rollkit/rollkit_test_util.go:3]: note: main now has this pulled out of the internal/ folder in main branch (not yet tagged)

Comment (damiannolan) [reply e2e/tests/rollkit/rollkit_test_util.go:3]: Thank you @Reecepbcups! Appreciate the update

**Comment by chatton** [comment]
closing this as this work is not being actively worked on right now, and enough changes have been made to the E2Es that this test will not work as is. We can use this PR as a reference when re-visiting this.

**Pull request closed without merging**